src/NUnitTestExtractor/NUnitTestExtractorApp.cs

```diff
@@ -72,70 +79,101 @@ public static Level ParseLevel(string levelToParse)
-        /// <param name="writer">The StreamWriter which will be passed to WriteTestDllAndName() </param>
-        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Reliability", "CA2001:AvoidCallingProblematicMethods", MessageId = "System.Reflection.Assembly.LoadFrom")]
-        public static void GetTests(IList<string> dlls, Level level)
+        /// <param name="assemblyFullPath"></param>
+        /// <returns></returns>
+        public static HashSet<string> GetInfoFromAssembly(string assemblyFullPath)
         {
-            if (dlls == null || dlls.Count == 0)
-            {
-                throw new ArgumentNullException("dlls", "no dlls were specified");
-            }
+            HashSet<string> rv = new HashSet<string>();
 
-            foreach (string dll in dlls)
+            foreach (var item in getValidTestSuiteTypesFromAssembly(assemblyFullPath))
             {
-                List<Type> testSuites = getValidTestSuites(dll);
+                Level lv = ParseLevel(_options.Level);
 
-                foreach (Type testSuite in testSuites)
+                switch (lv)
                 {
-                    getValidTestsFromTestSuite(testSuite, level);
+                    case Level.Namespace:
+                        rv.Add(item.Namespace);
+                        break;
+                    case Level.Class:
+                        rv.Add(item.FullName);
+                        break;
+                    case Level.Function:
+                        rv.IntersectWith(getValidTestsFromTestSuite(item));
+                        break;
+                    case Level.TestCase:
+                        break;
+                    default:
+                        throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, "Invalid level option: {0}", lv.ToString()));
                 }
             }
+
+            return rv;
         }
 
         /// <summary>
-        /// Get all valid test suites from the assembly
+        /// 
         /// </summary>
-        /// <param name="assemblyFullPath">The full qualified path for the assembly</param>
-        /// <returns>Return a list of test suites</returns>
-        private static List<Type> getValidTestSuites(string assemblyFullPath)
+        /// <param name="assemblyFullPath"></param>
+        /// <returns></returns>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Reliability", "CA2001:AvoidCallingProblematicMethods", MessageId = "System.Reflection.Assembly.LoadFrom")]
+        private static Assembly loadAssembly(string assemblyFullPath)
         {
             if (string.IsNullOrWhiteSpace(assemblyFullPath))
             {
-                throw new ArgumentNullException("Assembly full path can't be NULL or Empty.");
+                throw new ArgumentNullException("assemblyFullPath", "can't be NULL or Empty.");
```

Comment:
[Should] Replace these string checks with ThrowIf.StringIsNullOrWhiteSpace().  So:  `ThrowIf.StringIsNullOrWhiteSpace(assemblyFullPath, nameof(assemblyFullPath));`
instead of:
```
if (string.IsNullOrWhiteSpace(assemblyFullPath))
{
    throw new ArgumentNullException("assemblyFullPath", "can't be NULL or Empty.");
}
```

Reply:
OK